### PR TITLE
DOMTokenList.prototype.replace

### PR DESCRIPTION
**Pull request by UndefinedBehaviour**

### CODE REVIEW

**Comment by JakeChampion** [comment]
Thanks for the PR @UndefinedBehaviour it looks like the test suite failed in IE 8 and IE 9.

To run the tests locally you can run `node test/polyfills/server.js` and then open http://bs-local.com:9876/test?includePolyfills=yes&always=no&feature=DOMTokenList.prototype.replace in the browser you want to test.

```
Failures:
- ie/8.0:
   -> DOMTokenList.prototype.replace passes the feature detect
      undefined == true
- ie/9.0:
   -> DOMTokenList.prototype.replace passes the feature detect
      undefined == true
   -> DOMTokenList.prototype.replace should return false if this doesn't contain token
      false === true
   -> DOMTokenList.prototype.replace should throw a "SyntaxError" DOMException if token is the empty string
      false === true
   -> DOMTokenList.prototype.replace should throw a "SyntaxError" DOMException if newToken is the empty string
      false === true
   -> DOMTokenList.prototype.replace should throw an "InvalidCharacterError" DOMException if token contains whitespace
      false === true
   -> DOMTokenList.prototype.replace should throw an "InvalidCharacterError" DOMException if newToken contains whitespace
      false === true
   -> DOMTokenList.prototype.replace should replace token with newToken if this doesn't already contain newToken
      false === true
   -> DOMTokenList.prototype.replace should replace token with newToken if newToken precedes token in this's token set (simply remove token)
      false === true
   -> DOMTokenList.prototype.replace should replace token with newToken if newToken follows token in this's token set (move newToken to token's position)
      false === true
```

**Comment by UndefinedBehaviour** [comment]
@JakeChampion ie8 & 9 should be fixed now.

**Comment by UndefinedBehaviour** [comment]
old versions of Chrome.

**Review by JakeChampion — APPROVED** [review]

File: polyfills/DOMTokenList/prototype/replace/polyfill.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+			var tokenString = '' + token, newTokenString = '' + newToken;
+
+			try {
+				new DOMException();
+			} catch (e) {
+				self.DOMException = function (message, name) {
+					if (!(this instanceof DOMException)) return new DOMException(message, name);
+					this.message = message;
+					this.name = name;
+				}
```
Comment (JakeChampion) [inline polyfills/DOMTokenList/prototype/replace/polyfill.js:14]: Nit: We should likely add DOMException as a separate polyfill, from reading the spec, this is what I think the polyfill should be:
```js
self.DOMException = function DOMException(message, name) {
	this.message = String(message);
	if (name === undefined) {
		this.name = "Error";
	} else {
		this.name = name;
	}
	var error = Error(message);
	this.stack = error.stack;
	this.code = errorNamesToCode[name] || 0;
}
DOMException.prototype = Object.create(Error.prototype);
DOMException.prototype.constructor = DOMException;
var errorNamesToCode = {
	// Introduced in DOM Level 1:
	INDEX_SIZE_ERR: 1,
	DOMSTRING_SIZE_ERR: 2,
	HIERARCHY_REQUEST_ERR: 3,
	WRONG_DOCUMENT_ERR: 4,
	INVALID_CHARACTER_ERR: 5,
	NO_DATA_ALLOWED_ERR: 6,
	NO_MODIFICATION_ALLOWED_ERR: 7,
	NOT_FOUND_ERR: 8,
	NOT_SUPPORTED_ERR: 9,
	INUSE_ATTRIBUTE_ERR: 10,
	// Introduced in DOM Level 2:
	INVALID_STATE_ERR: 11,
	SYNTAX_ERR: 12,
	INVALID_MODIFICATION_ERR: 13,
	NAMESPACE_ERR: 14,
	INVALID_ACCESS_ERR: 15,
	// Introduced in DOM Level 3:
	VALIDATION_ERR: 16,
	TYPE_MISMATCH_ERR: 17,
	// From other specifications:
	SECURITY_ERR: 18,
	NETWORK_ERR: 19,
	ABORT_ERR: 20,
	URL_MISMATCH_ERR: 21,
	QUOTA_EXCEEDED_ERR: 22,
	TIMEOUT_ERR: 23,
	INVALID_NODE_TYPE_ERR: 24,
	DATA_CLONE_ERR: 25,
};
Object.keys(errorNamesToCode).forEach(function (n) {
	DOMException[n] = errorNamesToCode[n];
	DOMException.prototype[n] = errorNamesToCode[n];
});
```

**Pull request merged**